internal/database/event_logs.go

```diff
@@ -112,6 +112,7 @@ type EventLogStore interface {
 	// CountUsersWithSetting returns the number of users wtih the given temporary setting set to the given value.
 	CountUsersWithSetting(ctx context.Context, setting string, value any) (int, error)
 
+	// ❗ DEPRECATED: Use event recorders from internal/telemetryrecorder instead.
```

Comment:
```suggestion
	// DEPRECATED: Use event recorders from internal/telemetryrecorder instead.
```

Just this, in GoLand, has a visual indicator

<img width="206" alt="CleanShot 2023-09-22 at 13 09 06@2x" src="https://github.com/sourcegraph/sourcegraph/assets/2946214/53a57ed6-0f28-4e04-8abf-c7752e56266d">

Reply:
Unfortunately I'm pretty sure we have a linter that will hard-fail PRs that use "real" deprecated methods, which will probably be really annoying 😅 The emoji makes this more of a "soft" deprecation

Reply:
The linter only checks for new call sites, no?

We have 100+ docstrings with `// DEPRECATED:`

Reply:
Ah - even then, there may be legitimate uses for using the existing event logs directly still, as we're not quite GA.

I'll opened an issue to follow up with a hard deprecation after GA: https://github.com/sourcegraph/sourcegraph/issues/56973

Reply:
I guess after all this is fine, more like a FYI 😁

Reply:
vscode has it too 😎